admin_guide/custom_controllers.adoc

```diff
@@ -0,0 +1,300 @@
+[[admin-guide-custom-controllers]]
+= Creating Custom Controllers
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
+
+In the Kubernetes API, a controller is a control loop that watches the shared state of a cluster 
+through the API server and makes changes in an attempt to move the current state towards the desired state. 
+Examples of controllers that ship with {product-title} today are 
+xref:../architecture/core_concepts/deployments.adoc#replication-controllers[replication controllers] and 
+xref:../dev_guide/application_health.adoc#container-health-checks-using-probes[endpoints controllers].
+
+You can create custom controllers using the `oc observe` command. The commmand allows you to run script against a 
+cluster that reacts to changes in the cluster. The command is available as an image named `openshift/observe:latest`.
+
+//from man page
+On startup, `oc observe` lists all of the resources of a particular type and executes the provided script on each one. 
+The command watches the server for changes, and will re-execute the script for each update.
+
+The `oc observe` command works best for problems of the form _for every resource X, make sure Y is true_. Some
+examples of ways observe can be used include: 
+
+  * Ensuring every namespace has a quota or limit range object  
+  * Ensuring every service is registered in DNS by making calls to a DNS API  
+  * Sending an email alert whenever a node reports 'NotReady'  
+  * Watching for the 'FailedScheduling' event and write an IRC message  
+  * Dynamically provisioning persistent volumes when a new PVC is created  
+  * Deleting pods that have reached successful completion after a period of time.  
+
+=== Installing `oc observe`
+
+To obtain and run the `oc observe` command:
+
+. Run the following command:
++
+----
+docker run --entrypoint /bin/bash -it openshift/observe:latest
+
+/usr/bin/id: cannot find name for user ID 1001
+----
++
+This command pulls the `openshift/observe:latest` image, builds a container, and opens a bash shell to the container.
```

Comment:
"builds a container" is very confusing as this has nothing to do with `docker build`. Maybe "launches a container" is better.